### PR TITLE
Jetty 12.0.x re-enable client transport tests

### PR DESCRIPTION
**Pull request by sbordet**
The commits have been preserved to retain git "move" history (hopefully).

This PR re-enabled the client transport tests, and implements properly trailers in both client and server for all the transports.

This PR was a necessary first step to ensure that we actually run more tests to validate the Jetty 12 implementation for all transports.

The client transport tests were present in `jetty-ee9` and `jetty-ee10` as duplicates.
Those that were not strictly depending on the Servlet APIs have been moved to `jetty-core`.

A few of the client transport tests were disabled because some feature is still not implemented, such as handling of 100-Continue, but other PRs will re-enable those tests.

### CODE REVIEW

File: jetty-core/jetty-fcgi/jetty-fcgi-server/src/main/java/org/eclipse/jetty/fcgi/server/internal/HttpStreamOverFCGI.java [hunk 1]
```diff
@@ -106,7 +107,9 @@ else if (FCGI.Headers.SERVER_PROTOCOL.equalsIgnoreCase(name))
 
     public void onHeaders()
     {
-        String pathQuery = URIUtil.addPathQuery(_path, _query);
+        String pathQuery = _path;
+        if (!StringUtil.isBlank(_query))
+            pathQuery = URIUtil.addPathQuery(_path, _query);
```
Comment (joakime) [inline jetty-core/jetty-fcgi/jetty-fcgi-server/src/main/java/org/eclipse/jetty/fcgi/server/internal/HttpStreamOverFCGI.java]: Should probably just make URIUtil.addPathQuery smarter for blank.

It currently only tests for null, it should really be testing for isBlank() like you are doing here.

I added PR #8602 to fix URIUtil.addPathQuery

File: jetty-core/jetty-http2/jetty-http2-common/src/main/java/org/eclipse/jetty/http2/api/Stream.java [hunk 2]
```diff
@@ -119,6 +119,10 @@ public default CompletableFuture<Stream> push(PushPromiseFrame frame, Listener l
      * <p>{@link Stream.Data} objects may be stored away for later, asynchronous,
      * processing (for example, to process them only when all of them have been
      * received).</p>
+     * <p>Once the returned {@link Stream.Data} object indicates that the end
+     * of the read side of the stream has been reached, further calls to this
+     * method will return a {@link Stream.Data} object with the same indication,
+     * although the instance may be different).</p>
```
Comment (joakime) [inline jetty-core/jetty-http2/jetty-http2-common/src/main/java/org/eclipse/jetty/http2/api/Stream.java]: Rogue `)` here.

File: jetty-core/jetty-http2/jetty-http2-tests/src/test/java/org/eclipse/jetty/http2/tests/FlowControlStrategyTest.java [hunk 3]
```diff
@@ -258,13 +258,15 @@ public void onDataAvailable(Stream stream)
                             settings.put(SettingsFrame.INITIAL_WINDOW_SIZE, size);
                             stream.getSession().settings(new SettingsFrame(settings, false), Callback.NOOP);
                             // Do not release the data here.
+                            stream.demand();
```
Comment (joakime) [inline jetty-core/jetty-http2/jetty-http2-tests/src/test/java/org/eclipse/jetty/http2/tests/FlowControlStrategyTest.java:260]: This comment seems out of place now.

File: jetty-core/jetty-http3/jetty-http3-client-transport/src/main/java/org/eclipse/jetty/http3/client/transport/internal/HttpReceiverOverHTTP3.java [hunk 4]
```diff
@@ -99,6 +93,11 @@ public void onResponse(Stream.Client stream, HeadersFrame frame)
             {
                 if (LOG.isDebugEnabled())
                     LOG.debug("stalling response processing, no demand after headers on {}", this);
+
+                // TODO: If frame.isLast(), calling demand() on it will kinda of make little sense.
+                //  we should go back to notifySuccessRef being set before this block (to avoid the race)
+                //  and in receive() query it to call notifySuccess().
```
Comment (joakime) [inline jetty-core/jetty-http3/jetty-http3-client-transport/src/main/java/org/eclipse/jetty/http3/client/transport/internal/HttpReceiverOverHTTP3.java]: Open an issue about this?  Seems important.

File: jetty-core/jetty-http3/jetty-http3-common/src/main/java/org/eclipse/jetty/http3/internal/HTTP3Stream.java [hunk 5]
```diff
@@ -44,6 +48,10 @@ public abstract class HTTP3Stream implements Stream, CyclicTimeouts.Expirable, A
     private long idleTimeout;
     private long expireNanoTime;
     private Object attachment;
+    private boolean dataDemand;
+    private boolean dataStalled;
+    private boolean dataLast;
+    private boolean dataAvailable;
```
Comment (joakime) [inline jetty-core/jetty-http3/jetty-http3-common/src/main/java/org/eclipse/jetty/http3/internal/HTTP3Stream.java:54]: Some of these values once were part of the toString in a different class.
Can they be added to the toString() in this class?

File: jetty-core/jetty-http3/jetty-http3-common/src/main/java/org/eclipse/jetty/http3/internal/HTTP3StreamConnection.java [hunk 6]
```diff
@@ -442,7 +337,7 @@ private int fill(ByteBuffer byteBuffer)
     @Override
     public String toConnectionString()
     {
-        return String.format("%s[demand=%b,stalled=%b,parserDataMode=%b]", super.toConnectionString(), hasDemand(), isStalled(), parserDataMode);
+        return String.format("%s[dataMode=%b]", super.toConnectionString(), parser.isDataMode());
```
Comment (joakime) [inline jetty-core/jetty-http3/jetty-http3-common/src/main/java/org/eclipse/jetty/http3/internal/HTTP3StreamConnection.java]: Lost the debug on stalled.
Perhaps add the `stream` to this toString()?

File: jetty-core/jetty-server/src/test/java/org/eclipse/jetty/server/handler/ResourceHandlerTest.java [hunk 7]
```diff
@@ -98,6 +98,7 @@
  *
  * TODO: increase the testing going on here
  */
+@Disabled
```
Comment (joakime) [inline jetty-core/jetty-server/src/test/java/org/eclipse/jetty/server/handler/ResourceHandlerTest.java]: Oh no!

File: jetty-core/jetty-tests/jetty-test-client-transports/src/test/java/org/eclipse/jetty/test/client/transport/Continue100Test.java [hunk 8]
```diff
@@ -11,24 +11,9 @@
 // ========================================================================
 //
 
-package org.eclipse.jetty.ee10.http.client;
+package org.eclipse.jetty.test.client.transport;
 
-public enum Transport
+// TODO: write 100 Continue tests for Handler (not Servlet) functionality.
```
Comment (joakime) [inline jetty-core/jetty-tests/jetty-test-client-transports/src/test/java/org/eclipse/jetty/test/client/transport/Continue100Test.java:16]: Yes please.
Note that LocalConnectorTest reports that it has some 100-Continue tests. (looks suspect to me)

File: jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletContextHandler.java [hunk 9]
```diff
@@ -293,6 +293,12 @@ public ServletContextApi newServletContextApi()
         return new ServletContextApi();
     }
 
+    @Override
+    public InvocationType getInvocationType()
+    {
+        return InvocationType.BLOCKING;
```
Comment (joakime) [inline jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletContextHandler.java:299]: This seems unrelated.

File: jetty-ee10/jetty-ee10-tests/jetty-ee10-test-client-transports/src/test/java/org/eclipse/jetty/ee10/test/client/transport/BlockedIOTest.java [hunk 10]
```diff
@@ -11,76 +11,81 @@
...
-public class BlockedIOTest extends AbstractTest<TransportScenario>
+// TODO: this test fails because the second read blocks in another thread,
+//  but then the Servlet thread exits service(). Since the request has not
+//  been started async, the implementation tries to complete, but finds
+//  the blocked read as illegal (thrown from ChannelCallback.succeeded()).
+//  Throwing IllegalStateException from ChannelCallback causes a series
+//  of other IllegalStateException to be thrown.
+//  Need to find a better failure path, so the blocked read throws a proper
+//  IOException and the implementation is able to clean up properly.
+@Disabled
```
Comment (joakime) [inline jetty-ee10/jetty-ee10-tests/jetty-ee10-test-client-transports/src/test/java/org/eclipse/jetty/ee10/test/client/transport/BlockedIOTest.java:46]: Does this mean an implementation bug? or a test expectation mistake?

File: jetty-ee10/jetty-ee10-tests/jetty-ee10-test-client-transports/src/test/java/org/eclipse/jetty/ee10/test/client/transport/InformationalResponseTest.java [hunk 11]
```diff
@@ -33,49 +33,40 @@
...
-import static org.eclipse.jetty.http.client.Transport.FCGI;
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.contains;
 import static org.hamcrest.Matchers.is;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
-public class InformationalResponseTest extends AbstractTest<TransportScenario>
+// TODO: these tests fail with "last already written" since in core Jetty
+//  we don't handle well writing 2 HTTP responses one after the other.
+@Disabled
```
Comment (joakime) [inline jetty-ee10/jetty-ee10-tests/jetty-ee10-test-client-transports/src/test/java/org/eclipse/jetty/ee10/test/client/transport/InformationalResponseTest.java:47]: This a fault of the implementation on the server? or the client? 
Or is a problem in testcase assumptions/expectations?

File: jetty-ee10/jetty-ee10-tests/jetty-ee10-test-client-transports/src/test/java/org/eclipse/jetty/ee10/test/client/transport/ServerTimeoutsTest.java [hunk 12]
```diff
@@ -42,74 +42,61 @@
...
-
-public class ServerTimeoutsTest extends AbstractTest<TransportScenario>
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
+
+// TODO: the server-side timeout mechanism should be completely reworked.
+//  Currently, HttpChannel.onFailure() is called, but timeouts are different
+//  since they may be ignored, so we don't want to remember errors if they are ignored.
+//  However, this behavior is historically so because of Servlets, and we
+//  may decide differently for Handlers.
+@Disabled
```
Comment (joakime) [inline jetty-ee10/jetty-ee10-tests/jetty-ee10-test-client-transports/src/test/java/org/eclipse/jetty/ee10/test/client/transport/ServerTimeoutsTest.java:63]: Reads like a test assumption mistake?
But also that we need a version of this test fore core/handlers as well.

File: jetty-ee9/jetty-ee9-tests/jetty-ee9-test-client-transports/src/test/java/org/eclipse/jetty/ee9/test/client/transport/AsyncIOServletTest.java [hunk 13]
```diff
@@ -100,14 +93,52 @@
 import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.junit.jupiter.api.Assertions.fail;
 
-public class AsyncIOServletTest extends AbstractTest<AsyncIOServletTest.AsyncTransportScenario>
+// TODO: most of these tests do not work because the scope listener mechanism is broken.
+@Disabled
```
Comment (joakime) [inline jetty-ee9/jetty-ee9-tests/jetty-ee9-test-client-transports/src/test/java/org/eclipse/jetty/ee9/test/client/transport/AsyncIOServletTest.java:97]: Is this handled by the listener PR from @lorban ?

**Review by joakime — CHANGES_REQUESTED** [review]
Minor changes and questions

File: jetty-core/jetty-http2/jetty-http2-tests/src/test/java/org/eclipse/jetty/http2/tests/FlowControlStrategyTest.java (hunk 3 above)
Comment (sbordet) [reply jetty-core/jetty-http2/jetty-http2-tests/src/test/java/org/eclipse/jetty/http2/tests/FlowControlStrategyTest.java:260]: Why? We don't release, just demand.

File: jetty-core/jetty-http3/jetty-http3-client-transport/src/main/java/org/eclipse/jetty/http3/client/transport/internal/HttpReceiverOverHTTP3.java (hunk 4 above)
Comment (sbordet) [reply jetty-core/jetty-http3/jetty-http3-client-transport/src/main/java/org/eclipse/jetty/http3/client/transport/internal/HttpReceiverOverHTTP3.java]: It's a forgotten old comment that is not valid anymore. I'll just delete it, thanks for catching it.

File: jetty-core/jetty-http3/jetty-http3-common/src/main/java/org/eclipse/jetty/http3/internal/HTTP3Stream.java (hunk 5 above)
Comment (sbordet) [reply jetty-core/jetty-http3/jetty-http3-common/src/main/java/org/eclipse/jetty/http3/internal/HTTP3Stream.java:54]: Done.

File: jetty-core/jetty-http3/jetty-http3-common/src/main/java/org/eclipse/jetty/http3/internal/HTTP3StreamConnection.java (hunk 6 above)
Comment (sbordet) [reply jetty-core/jetty-http3/jetty-http3-common/src/main/java/org/eclipse/jetty/http3/internal/HTTP3StreamConnection.java]: Done.

File: jetty-core/jetty-server/src/test/java/org/eclipse/jetty/server/handler/ResourceHandlerTest.java (hunk 7 above)
Comment (sbordet) [reply jetty-core/jetty-server/src/test/java/org/eclipse/jetty/server/handler/ResourceHandlerTest.java]: Yes, needs to be looked at.

File: jetty-core/jetty-tests/jetty-test-client-transports/src/test/java/org/eclipse/jetty/test/client/transport/Continue100Test.java (hunk 8 above)
Comment (sbordet) [reply jetty-core/jetty-tests/jetty-test-client-transports/src/test/java/org/eclipse/jetty/test/client/transport/Continue100Test.java:16]: In another PR.

File: jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletContextHandler.java (hunk 9 above)
Comment (sbordet) [reply jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletContextHandler.java:299]: No, it is related because without it Servlets were invoked with PC in the selector thread, which would fail a bunch of client transport tests that were using just 1 selector.

File: jetty-ee10/jetty-ee10-tests/jetty-ee10-test-client-transports/src/test/java/org/eclipse/jetty/ee10/test/client/transport/BlockedIOTest.java (hunk 10 above)
Comment (sbordet) [reply jetty-ee10/jetty-ee10-tests/jetty-ee10-test-client-transports/src/test/java/org/eclipse/jetty/ee10/test/client/transport/BlockedIOTest.java:46]: Needs discussion with @gregw since the tests are legit IMHO (although it's bad code), and we should handle these cases better than what we do now.

File: jetty-ee10/jetty-ee10-tests/jetty-ee10-test-client-transports/src/test/java/org/eclipse/jetty/ee10/test/client/transport/InformationalResponseTest.java (hunk 11 above)
Comment (sbordet) [reply jetty-ee10/jetty-ee10-tests/jetty-ee10-test-client-transports/src/test/java/org/eclipse/jetty/ee10/test/client/transport/InformationalResponseTest.java:47]: It's a server implementation issue right now, since we assume that HTTP only has 1 response, while it can have many.

File: jetty-ee9/jetty-ee9-tests/jetty-ee9-test-client-transports/src/test/java/org/eclipse/jetty/ee9/test/client/transport/AsyncIOServletTest.java (hunk 13 above)
Comment (sbordet) [reply jetty-ee9/jetty-ee9-tests/jetty-ee9-test-client-transports/src/test/java/org/eclipse/jetty/ee9/test/client/transport/AsyncIOServletTest.java:97]: No, these are server-side listeners.

File: jetty-core/jetty-fcgi/jetty-fcgi-server/src/main/java/org/eclipse/jetty/fcgi/server/internal/HttpStreamOverFCGI.java (hunk 1 above)
Comment (joakime) [reply jetty-core/jetty-fcgi/jetty-fcgi-server/src/main/java/org/eclipse/jetty/fcgi/server/internal/HttpStreamOverFCGI.java]: PR #8602 has been merged to `jetty-12.0.x` now

File: jetty-core/jetty-http2/jetty-http2-tests/src/test/java/org/eclipse/jetty/http2/tests/FlowControlStrategyTest.java (hunk 3 above)
Comment (joakime) [reply jetty-core/jetty-http2/jetty-http2-tests/src/test/java/org/eclipse/jetty/http2/tests/FlowControlStrategyTest.java:260]: The other locations with the same pattern in this same file call demand, then make the comment about "do not release".
This specific one was the odd one out.
Having the comment above the demand made it seem like the demand was doing the "do not release".

File: jetty-core/jetty-http2/jetty-http2-common/src/main/java/org/eclipse/jetty/http2/api/Stream.java (hunk 2 above)
Comment (joakime) [reply jetty-core/jetty-http2/jetty-http2-common/src/main/java/org/eclipse/jetty/http2/api/Stream.java]: The opening `(` is either missing, or this `)` is a rogue one that should be deleted.

File: jetty-core/jetty-http3/jetty-http3-common/src/main/java/org/eclipse/jetty/http3/internal/HTTP3Stream.java [hunk 14]
```diff
@@ -146,10 +157,27 @@ public Data readData()
...
         {
-            HTTP3StreamConnection connection = (HTTP3StreamConnection)endPoint.getConnection();
-            Data data = connection.readData();
-            if (data != null)
-                updateClose(data.isLast(), false);
+            if (LOG.isDebugEnabled())
+                LOG.debug("reading data on {}", this);
+
+            Data data;
+            if (dataLast)
```
Comment (lorban) [inline jetty-core/jetty-http3/jetty-http3-common/src/main/java/org/eclipse/jetty/http3/internal/HTTP3Stream.java]: You're accessing `dataLast` outside lock scope.

File: jetty-ee10/jetty-ee10-tests/jetty-ee10-test-client-transports/src/test/java/org/eclipse/jetty/ee10/test/client/transport/ServerTimeoutsTest.java (hunk 12 above)
Comment (sbordet) [reply jetty-ee10/jetty-ee10-tests/jetty-ee10-test-client-transports/src/test/java/org/eclipse/jetty/ee10/test/client/transport/ServerTimeoutsTest.java:63]: The test assumption is correct, since this is a Servlet-based test.
I added a test class for core.

File: jetty-core/jetty-http3/jetty-http3-common/src/main/java/org/eclipse/jetty/http3/internal/HTTP3StreamConnection.java [hunk 15]
```diff
@@ -40,18 +40,13 @@ public abstract class HTTP3StreamConnection extends AbstractConnection
...
+    private HTTP3Stream stream;
+    private RetainableByteBuffer networkBuffer;
     private boolean applicationMode;
-    private boolean parserDataMode;
-    private boolean dataDemand;
-    private boolean dataStalled;
-    private DataFrame dataFrame;
-    private boolean dataLast;
-    private boolean noData;
     private boolean remotelyClosed;
```
Comment (lorban) [inline jetty-core/jetty-http3/jetty-http3-common/src/main/java/org/eclipse/jetty/http3/internal/HTTP3StreamConnection.java:50]: I'm not certain accesses to this class' fields are always thread-safe since the lock went away. Are you certain there are no concurrent accesses?

File: jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/AbstractConnector.java [hunk 16]
```diff
@@ -185,22 +185,14 @@ public AbstractConnector(
...
-                {
-                    pool = new LogarithmicArrayByteBufferPool();
-                    server.addBean(pool, true);
-                }
-                addBean(pool, false);
-            }
-            else
-            {
-                addBean(pool, true);
+                pool = new LogarithmicArrayByteBufferPool();
```
Comment (lorban) [inline jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/AbstractConnector.java:194]: You removed `addBean(pool, false)` - does that mean we don't allow configuring a pool on the connector anymore?

This sounds a bit out of scope for this PR to be honest.

File: jetty-core/jetty-server/src/test/java/org/eclipse/jetty/server/HttpServerTestBase.java [hunk 17]
```diff
@@ -1763,8 +1762,7 @@ public void onClosed(Connection connection)
         long total = contents.stream().mapToLong(Content.Chunk::remaining).sum();
         assertThat(total, equalTo(chunk.length * 4L));
 
-        ByteBufferPool bbp = _connector.getBean(ByteBufferPool.class);
-        RetainableByteBufferPool rbbp = bbp.asRetainableByteBufferPool();
+        RetainableByteBufferPool rbbp = _connector.getBean(RetainableByteBufferPool.class);
```
Comment (lorban) [inline jetty-core/jetty-server/src/test/java/org/eclipse/jetty/server/HttpServerTestBase.java:1765]: This code assumes a connector can have a `RetainableByteBufferPool`, but he above code assumes it cannot have a `ByteBufferPool`. This looks inconsistent.

File: jetty-core/jetty-tests/jetty-test-client-transports/src/test/java/org/eclipse/jetty/test/client/transport/AsyncRequestContentTest.java [hunk 18]
```diff
@@ -221,20 +205,14 @@ public void testBufferReuseAfterCallbackCompleted(Transport transport) throws Ex
...
         {
-            baseRequest.setHandled(true);
-            ServletInputStream input = request.getInputStream();
-            while (true)
-            {
-                int read = input.read();
-                if (read < 0)
-                    break;
-            }
+            Content.Source.consumeAll(request);
```
Comment (lorban) [inline jetty-core/jetty-tests/jetty-test-client-transports/src/test/java/org/eclipse/jetty/test/client/transport/AsyncRequestContentTest.java:213]: Wasn't reading one byte at a time done on purpose?

**Review by joakime — APPROVED** [review]

**Review by lorban — CHANGES_REQUESTED** [review]
Not fully reviewed yet.

File: jetty-core/jetty-http3/jetty-http3-common/src/main/java/org/eclipse/jetty/http3/internal/HTTP3Stream.java (hunk 14 above)
Comment (sbordet) [reply jetty-core/jetty-http3/jetty-http3-common/src/main/java/org/eclipse/jetty/http3/internal/HTTP3Stream.java]: Good catch! Fixed.

File: jetty-core/jetty-http3/jetty-http3-common/src/main/java/org/eclipse/jetty/http3/internal/HTTP3StreamConnection.java (hunk 15 above)
Comment (sbordet) [reply jetty-core/jetty-http3/jetty-http3-common/src/main/java/org/eclipse/jetty/http3/internal/HTTP3StreamConnection.java:50]: They are only accessed by the parser thread.

File: jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/AbstractConnector.java (hunk 16 above)
Comment (sbordet) [reply jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/AbstractConnector.java:194]: Now adding a `RetainableBBP` as a bean to the connector.

File: jetty-core/jetty-tests/jetty-test-client-transports/src/test/java/org/eclipse/jetty/test/client/transport/AsyncRequestContentTest.java (hunk 18 above)
Comment (sbordet) [reply jetty-core/jetty-tests/jetty-test-client-transports/src/test/java/org/eclipse/jetty/test/client/transport/AsyncRequestContentTest.java:213]: Don't think it was on purpose.

File: jetty-core/jetty-tests/jetty-test-client-transports/src/test/java/org/eclipse/jetty/test/client/transport/ServerTimeoutsTest.java [hunk 19]
```diff
@@ -11,28 +11,10 @@
...
+package org.eclipse.jetty.test.client.transport;
 
-import java.io.IOException;
-
-import org.junit.jupiter.api.AfterEach;
-
-public abstract class AbstractTest<T extends TransportScenario>
+// TODO: similar to eeX ServerTimeoutsTest but with Handler semantic.
+//  For example, we may decide to not ignore the timeouts if there is a thread dispatched to the Handler.
+public class ServerTimeoutsTest
```
Comment (lorban) [inline jetty-core/jetty-tests/jetty-test-client-transports/src/test/java/org/eclipse/jetty/test/client/transport/ServerTimeoutsTest.java:18]: Why not deleting this class instead of keeping it empty?

File: jetty-ee10/jetty-ee10-tests/jetty-ee10-test-client-transports/src/test/java/org/eclipse/jetty/ee10/test/client/transport/AsyncIOServletTest.java [hunk 20]
```diff
@@ -100,14 +94,52 @@
 import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.junit.jupiter.api.Assertions.fail;
 
-public class AsyncIOServletTest extends AbstractTest<AsyncIOServletTest.AsyncTransportScenario>
+// TODO: most of these tests do not work because the scope listener mechanism is broken.
+@Disabled
```
Comment (lorban) [inline jetty-ee10/jetty-ee10-tests/jetty-ee10-test-client-transports/src/test/java/org/eclipse/jetty/ee10/test/client/transport/AsyncIOServletTest.java:98]: Maybe add the issue # (or the PR #) in the comment?

**Review by lorban — APPROVED** [review]
A couple minor nits, otherwise LGTM.

File: jetty-core/jetty-tests/jetty-test-client-transports/src/test/java/org/eclipse/jetty/test/client/transport/ServerTimeoutsTest.java (hunk 19 above)
Comment (sbordet) [reply jetty-core/jetty-tests/jetty-test-client-transports/src/test/java/org/eclipse/jetty/test/client/transport/ServerTimeoutsTest.java:18]: To remember that tests needs to be written.
